fix(identity): M0 and M1 reference page updates from feedback (#5747)
Kong/developer.konghq.com: app/_includes/plugins/consumers-vs-principals.md

The {{ include.name }} plugin supports two types of authenticating entities: [Consumers](/gateway/entities/consumer/) and [principals](/identity/principals/) {% new_in 3.15 %}.

The following table will help you decide which to use based on your use case:

<!--vale off-->
{% table %}
columns:
  - title: Use case
    key: usecase
  - title: Choose
    key: choose
rows:

>   - usecase: You are running on-prem or {{site.konnect_short_name}} and need authentication plugins other than Key Auth or Basic Auth.

Reviewer: I didn't quite follow the second part of this sentence. Perhaps we can simplify though? If you're running on-prem Principals won't work, you _have_ to use Consumers.

>   - usecase: You are running on-prem or {{site.konnect_short_name}} and need authentication plugins other than Key Auth or Basic Auth.

Reviewer 2: ```suggestion
  - usecase: You are running on-prem and need authentication plugins.
```

    choose: "[Consumers](/gateway/entities/consumer/)"

>   - usecase: You need to scope plugins directly to an authenticating entity.

Reviewer: This can be done for Principals too, using conditional expressions. Let's skip this.

But I'd suggest _adding_ a use case of "I have a highly performance sensitive application that cannot be dependent on connectivity between my Gateway data planes and Konnect.", in which case they should choose Consumers.

>   - usecase: You need to scope plugins directly to an authenticating entity.

Reviewer 2: ```suggestion
  - usecase: You have a highly performance-sensitive application that can't rely on connectivity between your Gateway data planes and {{site.konnect_short_name}}.
```

    choose: "[Consumers](/gateway/entities/consumer/)"
  - usecase: You are on {{site.konnect_short_name}} and need to share identity across multiple control planes in the same region without manually syncing credentials.
    choose: "[Principals](/identity/principals/)"

>   - usecase: You have a large population of authenticating entities that would exceed [Consumer control plane limits](/gateway/control-plane-resource-limits/). Consumers are loaded into data plane memory; principals are loaded on demand.

Reviewer: Suggest adding a bit of detail: "loaded on demand from Konnect and cached in the data plane."

>   - usecase: You have a large population of authenticating entities that would exceed [Consumer control plane limits](/gateway/control-plane-resource-limits/). Consumers are loaded into data plane memory; principals are loaded on demand.

Reviewer 2: ```suggestion
  - usecase: You have a large population of authenticating entities that would exceed [Consumer control plane limits](/gateway/control-plane-resource-limits/). {{site.konnect_short_name}} loads Consumers into data plane memory, but loads principals on demand and caches them in the data plane.
```
